### PR TITLE
Resolve plotly.py dependency issue

### PR DESCRIPTION
**Pull request by knshnb**
## Contributor License Agreement

This repository (``optuna-dashboard``) and [Goptuna](https://github.com/c-bata/goptuna) share common code.
This pull request may therefore be ported to Goptuna.
Make sure that you understand the consequences concerning licenses and check the box below if you accept the term before creating this pull request.

* [x] I agree this patch may be ported to [Goptuna](https://github.com/c-bata/goptuna) by other Goptuna contributors.

## Reference Issues/PRs

Follow-up of https://github.com/optuna/optuna-dashboard/pull/764

## What does this implement/fix? Explain your changes.

- Add plotly to the optional dependency
- Fall back to plotly.js when plotly.py is not available

### CODE REVIEW

File: optuna_dashboard/ts/components/GraphContour.tsx [hunk 1]
```diff
@@ -17,15 +17,25 @@ import { useMergedUnionSearchSpace } from "../searchSpace"
...
   if (query.get("plotlypy_rendering") === "true") {
-    return <ContourBackend study={study} />
+    if (plotlypyIsAvailable) {
+      return <ContourBackend study={study} />
+    } else {
+      console.warn(
+        "Use frontend rendering because plotlypy is specified but not available."
+      )
+      return <ContourFrontend study={study} />
+    }
```
Comment (c-bata) [inline optuna_dashboard/ts/components/GraphContour.tsx:38]: Perhaps, we could define the following wrapper function in `state.ts`:

```js
export const usePlotlypyIsAvailable = (): boolean => {
  const query = useQuery()
  const plotlypyIsAvailable = useRecoilValue<boolean>(plotlypyIsAvailableState)

  if (query.get("plotlypy_rendering") === "true") {
    if (plotlyPyIsAvailable) {
      return true
    }
    console.warn(...)
  }
  return false
}
```

**Review by c-bata — APPROVED** [review]
Looks good to me. I left one suggestion though.

File: optuna_dashboard/ts/components/GraphContour.tsx (hunk 1 above)
Comment (c-bata) [reply optuna_dashboard/ts/components/GraphContour.tsx:38]: Furthermore, we could use notistack to show a warning message to users like below:
https://github.com/optuna/optuna-dashboard/blob/8033e79da6de5a4a9036a9375083ed93cc291772/optuna_dashboard/ts/action.ts#L234-L236

Comment (knshnb) [reply optuna_dashboard/ts/components/GraphContour.tsx:38]: Yeah, I will introduce this function when I implement backend rendering of  other graphs 👍

**Comment by knshnb** [comment]
Thanks for the review! I worked on https://github.com/optuna/optuna-dashboard/pull/773#discussion_r1467387250 by another PR: https://github.com/optuna/optuna-dashboard/pull/778.

**Pull request merged**